src/EditorFeatures/CSharp/DecompiledSource/CSharpDecompiledSourceFormattingRule.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Formatting.Rules;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.DecompiledSource
+{
+    internal class CSharpDecompiledSourceFormattingRule : AbstractFormattingRule
+    {
+        public static readonly AbstractFormattingRule Instance = new CSharpDecompiledSourceFormattingRule();
+
+        private CSharpDecompiledSourceFormattingRule()
+        {
+        }
+
+        public override AdjustNewLinesOperation GetAdjustNewLinesOperation(
+            SyntaxToken previousToken, SyntaxToken currentToken, AnalyzerConfigOptions options, in NextGetAdjustNewLinesOperation nextOperation)
+        {
+            var operation = GetAdjustNewLinesOperation(previousToken, currentToken);
+            return operation ?? nextOperation.Invoke();
+        }
+
+        private AdjustNewLinesOperation GetAdjustNewLinesOperation(SyntaxToken previousToken, SyntaxToken currentToken)
+        {
+            // To help code not look too tightly packed, we place a blank line after every statement that ends with a
+            // `}` (unless it's also followed by another `}`).
```

Comment:
What about `else if`?

Reply:
i can add comment, but that will be ok.  the `else` clause is not a statement itself.  so there won't be a blank line in:

```c#
if (foo)
{
}

else
```